### PR TITLE
Add Firebase Crashlytics

### PR DESCRIPTION
**Pull request by mathewscullin**
## Overview

Added Firebase Crashlytics to the code base. Per TestFlight notes, modified our presenting of currentVC when opening the app so that the HomeViewController doesn't animate modally once a user has signed in. Also changed the initial screen background color to `CourseGrabBlack`.

## Changes Made

^^

## Test Coverage

Crashed the app to ensure Firebase logged it  and tested on the simulator trying out all possible cases for app launch.

### CODE REVIEW

File: CourseGrab/SceneDelegate.swift [hunk 1]
```diff
@@ -18,11 +18,11 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
         let window = UIWindow(windowScene: scene)
         let splash = UIViewController()
-        splash.view.backgroundColor = .white
+        splash.view.backgroundColor = .courseGrabBlack
         window.rootViewController = splash
 
         Auth.auth().addStateDidChangeListener { (auth, user) in
-            if let email = user?.email, email.split(separator: "@").last != "cornell.edu" {
+            if let email = user?.email, email.split(separator: "@").last != "cornell.edu" && email != "coursegrabappstore@gmail.com" {
```
Comment (danielVebman) [inline CourseGrab/SceneDelegate.swift:25]: Do we still need this hard coded?

File: CourseGrab/SceneDelegate.swift [hunk 2]
```diff
@@ -48,10 +48,10 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                     // If both are LoginViewController, don't bother presenting a new VC
                     return
                 }
-
+                
                 newVC.modalPresentationStyle = .overFullScreen
                 newVC.modalPresentationCapturesStatusBarAppearance = true
-                currentVC.present(newVC, animated: true)
+                currentVC.present(newVC, animated: user == nil || currentVC is LoginViewController)
```
Comment (danielVebman) [inline CourseGrab/SceneDelegate.swift:54]: Maybe add a brief comment here on why we have this condition.

**Review by danielVebman — APPROVED** [review]
Looks good. Just two minor comments.

File: CourseGrab/SceneDelegate.swift [hunk 3]
```diff
@@ -48,10 +48,11 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                     // If both are LoginViewController, don't bother presenting a new VC
                     return
                 }
-
+                
                 newVC.modalPresentationStyle = .overFullScreen
                 newVC.modalPresentationCapturesStatusBarAppearance = true
-                currentVC.present(newVC, animated: true)
+                // Only present LoginViewController modally
```
Comment (danielVebman) [inline CourseGrab/SceneDelegate.swift:54]: Is this accurate? It will also display HomeVC modally over LoginVC, no?

**Review by danielVebman — APPROVED** [review]

File: CourseGrab/SceneDelegate.swift (hunk 3 above)
Comment (mathewscullin) [reply CourseGrab/SceneDelegate.swift:54]: Yup, in that case we want it to do that. Think of it as a session, when you sign in or sign out, the user will enter or exit the session modally but once you're in even if you close the app on your phone it shouldn't present modally.

File: CourseGrab/SceneDelegate.swift (hunk 1 above)
Comment (danielVebman) [reply CourseGrab/SceneDelegate.swift:25]: I was talking about the coursegrabappstore email.

Comment (mathewscullin) [reply CourseGrab/SceneDelegate.swift:25]: My bad, I just added it back in to reflect whats currently on the app store but its probably better to take it out.

**Review by ywsang — APPROVED** [review]
lgtm!

File: CourseGrab/SceneDelegate.swift [hunk 4]
```diff
@@ -48,10 +48,11 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                     // If both are LoginViewController, don't bother presenting a new VC
                     return
                 }
-
+
```
Comment (Omarrasheed) [inline CourseGrab/SceneDelegate.swift:51]: We should setup swiftlint

**Review by Omarrasheed — APPROVED** [review]

**Pull request merged**